### PR TITLE
Add support for ExtensionKit extensions

### PR DESCRIPTION
**Pull request by tovkal**
Resolves https://github.com/tuist/tuist/issues/4941

### Short description 📝

I needed to add an App Intent extension and found that support for [ExtensionKit extensions](https://developer.apple.com/documentation/extensionkit) was not yet implemented in tuist.

I added and modified unit tests and also modified an acceptance test, but this is my first time contributing to tuist so let me know if something is missing.

### How to test the changes locally 🧐

Run `./fourier test tuist acceptance projects/tuist/features/generate-4.feature:47`

### Contributor checklist ✅

- [x] The code has been linted using run `./fourier lint tuist --fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Contributors have checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Review by danyf90 — APPROVED** [review]
Looks great to me! 👏 👏 👏

**Comment by danyf90** [comment]
@allcontributors add @tovkal for code

File: projects/tuist/fixtures/ios_app_with_extensions/Project.swift [hunk 1]
```diff
@@ -106,5 +107,13 @@ let project = Project(
             infoPlist: .default,
             sources: "StaticFramework/Sources/**"
         ),
+        Target(
+            name: "AppIntentExtension",
+            platform: .iOS,
+            product: .extensionKitExtension,
+            bundleId: "io.tuist.App.AppIntentExtension",
+            infoPlist: .default,
```
Comment (kwridan) [inline projects/tuist/fixtures/ios_app_with_extensions/Project.swift:115]: App intents extension need this extra key 

```suggestion
            infoPlist: .extendingDefault(with: [
                "EXAppExtensionAttributes": [
                    "EXExtensionPointIdentifier": "com.apple.appintents-extension"
                ]
            ]),
```

I spotted this when manually creating a new app intents extension in Xcode

app-intents-plist

**Review by kwridan — APPROVED** [review]
This is a great first contribution @tovkal ! 👏

I believe you've covered all the cases, from my testing it matches the behaviour I've observed in Xcode when manually creating the new extension.

Left a minor comment for the fixture

**Comment by tovkal** [comment]
Seems like the `ios_app_with_extensions` scenario (the one I modified) is failing to build? I rested on my machine and works fine there 😕

**Comment by danyf90** [comment]
> Seems like the `ios_app_with_extensions` scenario (the one I modified) is failing to build? I rested on my machine and works fine there 😕

The error seems to be in the test, you can replicate it with `./fourier test tuist acceptance projects/tuist/features/generate-4.feature`:
```
  Scenario: The project is a tvOS application with extensions (tvos_app_with_extensions)                        # projects/tuist/features/generate-4.feature:59
    Given that tuist is available                                                                               # projects/tuist/features/step_definitions/shared/tuist.rb:4
      swift build -c release --product tuist --package-path /private/tmp/tuist
      swift build -c release --product ProjectDescription --package-path /private/tmp/tuist
      swift build -c release --product tuistenv --package-path /private/tmp/tuist
    And I have a working directory                                                                              # projects/tuist/features/step_definitions/shared/workspace.rb:6
    Then I copy the fixture tvos_app_with_extensions into the working directory                                 # projects/tuist/features/step_definitions/shared/fixtures.rb:5
    Then tuist generates the project                                                                            # projects/tuist/features/step_definitions/shared/tuist.rb:30
      /private/tmp/tuist/.build/release/tuist generate --no-cache --no-open --path /var/folders/80/zjd47vsn6csfsmhzgb69qvj00000gp/T/d20230129-21620-vlcwg
    Then I should be able to build for tvOS the scheme App                                                      # projects/tuist/features/step_definitions/shared/xcode.rb:6
      undefined method `flunk' for Xcode:Module (NoMethodError)
      ./projects/tuist/features/support/xcode.rb:128:in `valid_simulator_destination_for_platform'
      ./projects/tuist/features/step_definitions/shared/xcode.rb:17:in `/I should be able to (.+) for (iOS|macOS|tvOS|watchOS) the scheme (.+)/'
      projects/tuist/features/generate-4.feature:64:in `I should be able to build for tvOS the scheme App'
```

I think the error is misleading and there is a problem somewhere else in the part you have changed, but everything seems fine 🤔 🤷‍♂️

**Comment by tovkal** [comment]
> > Seems like the `ios_app_with_extensions` scenario (the one I modified) is failing to build? I rested on my machine and works fine there 😕
> 
> The error seems to be in the test, you can replicate it with `./fourier test tuist acceptance projects/tuist/features/generate-4.feature`:
> 
> ```
>   Scenario: The project is a tvOS application with extensions (tvos_app_with_extensions)                        # projects/tuist/features/generate-4.feature:59
>     Given that tuist is available                                                                               # projects/tuist/features/step_definitions/shared/tuist.rb:4
>       swift build -c release --product tuist --package-path /private/tmp/tuist
>       swift build -c release --product ProjectDescription --package-path /private/tmp/tuist
>       swift build -c release --product tuistenv --package-path /private/tmp/tuist
>     And I have a working directory                                                                              # projects/tuist/features/step_definitions/shared/workspace.rb:6
>     Then I copy the fixture tvos_app_with_extensions into the working directory                                 # projects/tuist/features/step_definitions/shared/fixtures.rb:5
>     Then tuist generates the project                                                                            # projects/tuist/features/step_definitions/shared/tuist.rb:30
>       /private/tmp/tuist/.build/release/tuist generate --no-cache --no-open --path /var/folders/80/zjd47vsn6csfsmhzgb69qvj00000gp/T/d20230129-21620-vlcwg
>     Then I should be able to build for tvOS the scheme App                                                      # projects/tuist/features/step_definitions/shared/xcode.rb:6
>       undefined method `flunk' for Xcode:Module (NoMethodError)
>       ./projects/tuist/features/support/xcode.rb:128:in `valid_simulator_destination_for_platform'
>       ./projects/tuist/features/step_definitions/shared/xcode.rb:17:in `/I should be able to (.+) for (iOS|macOS|tvOS|watchOS) the scheme (.+)/'
>       projects/tuist/features/generate-4.feature:64:in `I should be able to build for tvOS the scheme App'
> ```
> 
> I think the error is misleading and there is a problem somewhere else in the part you have changed, but everything seems fine 🤔 🤷‍♂️

That is what I thought, but on my machine the test pass correctly. I also tried to run tuist directly on the fixture directory and the generated project compiles correctly. So I don't know why it's failing. Could you reproduce the test failure in your machine?

**Comment by danyf90** [comment]
> > > Seems like the `ios_app_with_extensions` scenario (the one I modified) is failing to build? I rested on my machine and works fine there 😕
> 
> > 
> 
> > The error seems to be in the test, you can replicate it with `./fourier test tuist acceptance projects/tuist/features/generate-4.feature`:
> 
> > 
> 
> > ```
> 
> >   Scenario: The project is a tvOS application with extensions (tvos_app_with_extensions)                        # projects/tuist/features/generate-4.feature:59
> 
> >     Given that tuist is available                                                                               # projects/tuist/features/step_definitions/shared/tuist.rb:4
> 
> >       swift build -c release --product tuist --package-path /private/tmp/tuist
> 
> >       swift build -c release --product ProjectDescription --package-path /private/tmp/tuist
> 
> >       swift build -c release --product tuistenv --package-path /private/tmp/tuist
> 
> >     And I have a working directory                                                                              # projects/tuist/features/step_definitions/shared/workspace.rb:6
> 
> >     Then I copy the fixture tvos_app_with_extensions into the working directory                                 # projects/tuist/features/step_definitions/shared/fixtures.rb:5
> 
> >     Then tuist generates the project                                                                            # projects/tuist/features/step_definitions/shared/tuist.rb:30
> 
> >       /private/tmp/tuist/.build/release/tuist generate --no-cache --no-open --path /var/folders/80/zjd47vsn6csfsmhzgb69qvj00000gp/T/d20230129-21620-vlcwg
> 
> >     Then I should be able to build for tvOS the scheme App                                                      # projects/tuist/features/step_definitions/shared/xcode.rb:6
> 
> >       undefined method `flunk' for Xcode:Module (NoMethodError)
> 
> >       ./projects/tuist/features/support/xcode.rb:128:in `valid_simulator_destination_for_platform'
> 
> >       ./projects/tuist/features/step_definitions/shared/xcode.rb:17:in `/I should be able to (.+) for (iOS|macOS|tvOS|watchOS) the scheme (.+)/'
> 
> >       projects/tuist/features/generate-4.feature:64:in `I should be able to build for tvOS the scheme App'
> 
> > ```
> 
> > 
> 
> > I think the error is misleading and there is a problem somewhere else in the part you have changed, but everything seems fine 🤔 🤷‍♂️
> 
> 
> 
> That is what I thought, but on my machine the test pass correctly. I also tried to run tuist directly on the fixture directory and the generated project compiles correctly. So I don't know why it's failing. Could you reproduce the test failure in your machine?

Yes I can replicate it with the fourier command

**Comment by danyf90** [comment]
@kwridan @fortmarek do you have any idea why fourier is failing? Codewise looks good to me 🤔

**Comment by fortmarek** [comment]
Currently, it looks like the code doesn't compile. Once the failure is only in `fourier`, I can try and have a look.

**Comment by danyf90** [comment]
@fortmarek just to add on this:
- `./fourier test tuist acceptance projects/tuist/features/generate-4.feature:47` works fine
- `./fourier test tuist acceptance projects/tuist/features/generate-4.feature` fails with error:
```
…
    Then tuist generates the project                                                                            # projects/tuist/features/step_definitions/shared/tuist.rb:30
      /private/tmp/tuist/.build/release/tuist generate --no-cache --no-open --path /var/folders/80/zjd47vsn6csfsmhzgb69qvj00000gp/T/d20230205-74588-evufef
    Then I should be able to build for tvOS the scheme App                                                      # projects/tuist/features/step_definitions/shared/xcode.rb:6
      undefined method `flunk' for Xcode:Module (NoMethodError)
      ./projects/tuist/features/support/xcode.rb:128:in `valid_simulator_destination_for_platform'
      ./projects/tuist/features/step_definitions/shared/xcode.rb:17:in `/I should be able to (.+) for (iOS|macOS|tvOS|watchOS) the scheme (.+)/'
      projects/tuist/features/generate-4.feature:64:in `I should be able to build for tvOS the scheme App'
```

**Comment by fortmarek** [comment]
Hey @tovkal 👋 

Can you cherry-pick [this](https://github.com/tuist/tuist/pull/5031/commits/d2f6ee0d838254bb4422fbcf8dc2e17c21d15f05) commit? The flunk issue is no longer there but it seems there is still some valid issue with this. Let me know if you need any further help!

**Comment by tovkal** [comment]
Done. Now it's failing for me in my computer (it wasn't before). As before in Actions, it is failing in `tvos_app_with_extensions`, which I did not change 🤔. I will work on it tomorrow, see if I can find the problem

**Comment by tovkal** [comment]
Seems like the error I had was because tvOS 16.1 got messed up (Xcode was offering to install it while at the same time showing it as installed). Reinstalling fixed the issue for me.

I see on the last Actions run that it's failing to build `ios_app_with_extensions`, but that passes correctly for me. No idea what difference in setup makes the test fail in Actions but pass on my machine. Tried to clone the repo again, but same thing there.

**Comment by danyf90** [comment]
This is still blocked by the Xcode 14 PR, I'll try to work on that if I find some time 🙏

**Comment by danyf90** [comment]
Keeping open, waiting for Xcode 14 PR

**Comment by danyf90** [comment]
Xcode 14 PR merged 🤞

**Comment by danyf90** [comment]
@tovkal could you please fix the conflicts and have a look at the CI failures? 🙏

**Comment by tovkal** [comment]
The check for that runs the test for `projects/tuist/features/generate-2.feature` seems to have failed in `Given that tuist is available`. As far as I am aware I have not modified anything that impacts the tests in `generate-2.feature`. `generate-2.feature` also succeeds in my machine. Maybe it is a bit flaky? 

I have updated the branch with `main` and solved conflicts.

**Comment by apps4everyone** [comment]
is this already in a release? Is there a release the next weeks? Thx

**Pull request merged**